src/server/pages/revisions-page.jsx

```diff
@@ -136,12 +136,9 @@ export default class RevisionsPage extends React.Component {
 
   onDeleteRevision(revisionID) {
     if (this.state.revisions.length === 1) {
-      fetchWithCSRFTokenAndJSONContent(
-        `/api/v1/notebooks/${this.props.ownerInfo.notebookId}/`,
-        {
-          method: "DELETE"
-        }
-      ).then(this.onDeleteNotebook);
+      deleteNotebookRequest(this.props.ownerInfo.notebookId).then(
```

Comment:
we are making a point to make functions handling promises be async by default so they're a bit easier to read. Could you make this `async onDeleteRevision` & then deal with the promise w/ `await`?